ntpd/src/daemon/spawn/mod.rs

```diff
@@ -272,24 +259,43 @@ where
         action_tx: mpsc::Sender<SpawnEvent>,
         mut system_notify: mpsc::Receiver<SystemEvent>,
     ) -> Result<(), E> {
-        use tokio::time::{timeout, Duration};
-        // basic event loop where init is called on startup and then wait for
-        // events from the system before doing anything
-        self.handle_init(&action_tx).await?;
-        while let Some(event) = timeout(Duration::from_secs(60), system_notify.recv())
-            .await
-            .unwrap_or(Some(SystemEvent::Idle))
-        {
+        let mut has_ticket = true;
+        let mut last_ticket_time = Instant::now();
+
+        loop {
+            if Instant::now() - last_ticket_time >= NETWORK_WAIT_PERIOD {
```

Comment:
I think `last_ticket_time.elapsed() >= NETWORK_WAIT_PERIOD` is easier to understand.

Reply:
done